e2e/tests/components/tabs/default.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { expect, test } from '../../../fixtures'
+
+test.describe('tabs, default', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/components/tabs/default')
+  })
+
+  test.describe('@desktop', () => {
+    test('displays four tabs', async ({ page }) => {
+      await expect(page.getByRole('tab')).toHaveCount(4)
+    })
+
+    test('displays the contents of one tab', async ({ page }) => {
+      await expect(page.getByRole('tabpanel')).toHaveCount(1)
+      await expect(page.getByRole('table')).toHaveCount(1)
+    })
+  })
+
+  test.describe('@mobile', () => {
+    test('does not display any tabs', async ({ page }) => {
+      await expect(page.getByRole('tab')).toHaveCount(0)
+      await expect(page.getByRole('tabpanel')).toHaveCount(0)
+    })
+
+    test('displays the contents list', async ({ page }) => {
+      await expect(page.getByRole('heading', { name: 'Contents' })).toBeVisible()
+      await expect(page.getByRole('listitem')).toHaveCount(4)
+    })
+
+    test('displays four tables', async ({ page }) => {
+      await expect(page.getByRole('table')).toHaveCount(4)
+    })
+  })
+
+  test.describe('when JavaScript is disabled', () => {
+    test.use({ javaScriptEnabled: false })
+
+    test('does not display any tabs', async ({ page }) => {
+      await expect(page.getByRole('tab')).toHaveCount(0)
+      await expect(page.getByRole('tabpanel')).toHaveCount(0)
+    })
+
+    test('displays the contents list', async ({ page }) => {
+      await expect(page.getByRole('heading', { name: 'Contents' })).toBeVisible()
+      await expect(page.getByRole('listitem')).toHaveCount(4)
+    })
+
+    test('displays four tables', async ({ page }) => {
+      await expect(page.getByRole('table')).toHaveCount(4)
+    })
+
+    test.describe('@visual-regression', () => {
+      test('matches the saved screenshot', async ({ componentElement }) => {
+        await expect(componentElement).toHaveScreenshot('no-javascript.png')
```

Comment:
Just a thought.
I wonder if this should be sent to the visual regression folder. Just a bit concerned that will could have snapshots in different locations.

Reply:
I was thinking we could just run the visual regression tests twice, once with JavaScript on and once with it off.

But there'll still be a few odd cases where we want to do a visual regression test after an interaction (hover, click etc.)

Reply:
Another option: we could configure [`snapshotPathTemplate`](https://playwright.dev/docs/api/class-testconfig#test-config-snapshot-path-template) so that they are all in one directory.

Reply:
yes let go with that